### ISSUE TITLE
Broken links using .html extension

### ISSUE THREAD

**Issue by imteekay** [wwsalmon/portfolio#59]
Apparently, your website replaced the .html extension with a URL without it.
Some of the links from old posts are still linking to the version with the .html extension.

One example is the [Understanding the math behind neural networks by building one from scratch (no TF/Keras, just numpy)](https://www.samsonzhang.com/2020/11/24/understanding-the-math-behind-neural-networks-by-building-one-from-scratch-no-tf-keras-just-numpy) post linking to the [visualization of gradient descent here](https://www.samsonzhang.com/2020/09/27/how-excel-calculates-trendlines-and-how-you-can-too-an-introduction-to-gradient-descent-the-most-important-machine-learning-algorithm) (but with the .html extension).
